stakepooldclient/stakepooldclient.go

```diff
@@ -731,3 +734,41 @@ func (s *stakepooldManager) CrossCheckColdWalletExtPubs(ctx context.Context, dcr
 	}
 	return nil
 }
+
+func (s *stakepooldManager) GetFeeAddress(ctx context.Context, txHash *chainhash.Hash, signature string) (*pb.GetFeeAddressResponse, error) {
+	var err error
+	responses := make([]*pb.GetFeeAddressResponse, len(s.grpcConnections))
+	for i, conn := range s.grpcConnections {
+		client := pb.NewStakepooldServiceClient(conn)
+		responses[i], err = client.GetFeeAddress(ctx, &pb.GetFeeAddressRequest{
+			Hash:      txHash.String(),
+			Signature: signature,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("GetFeeAddress RPC failed on stakepoold instance %s: %v", conn.Target(), err)
+		}
+	}
+	return responses[0], nil
```

Comment:
2 questions here:
- This method errors if any of the voting wallets do not return an address but the addresses returned from all backends are not compared for consistency. If it doesn't matter whether they all return the same address, I suppose it should likewise not matter if 1 or more backends do not return an address.
- If a previous call to this method ends up deriving a new address for some wallets and not others, a new call to the method would produce different addresses. I expect the wallet that failed previously would be an index behind and would not be tracking usage for the newer address. I think that just means the balance may not be accurate on that wallet, so no biggie perhaps. If no biggie, could just ignore `GetFeeAddress RPC failure` on any failing backend.

I still think using a cold wallet's xpub is better anyways.

Reply:
sql going away

Reply:
🤔 reply meant for another comment perhaps?